nginx_stage/spec/nginx_stage_config_spec.rb

```diff
@@ -0,0 +1,20 @@
+require 'nginx_stage'
+require 'spec_helper'
+
+describe NginxStage::Configuration do
+  describe 'does config match with example file?' do
+    it 'returns true if config matches example file' do
+      public_inst_methods = NginxStage::Configuration.public_instance_methods
+      attr_setter_names = public_inst_methods.map(
+          &:to_s).select { |name| name.end_with?("=") }.uniq
+
+      config_opts = attr_setter_names.map { |name| name.delete("=") } 
+      example_config_opts = File.read(
+        './share/nginx_stage_example.yml').scan(/#[\w_]+:/)
+
+      config_opts.each do |opt|
+        expect(example_config_opts).to opt_exists?(opt)
```

Comment:
Like below, you are just comparing arrays here and there is no need for a custom matcher. If you think the output is really egregious you could do an xor of the two arrays:

```
irb(main):048:0> a1 = ["one", "two", "three"]
=> ["one", "two", "three"]
irb(main):049:0> a2 = ["two", "three", "four"]
=> ["two", "three", "four"]
irb(main):050:0> a1 - a2
=> ["one"]
irb(main):051:0> a2 - a1
=> ["four"]
irb(main):052:0> a1 & a2
=> ["two", "three"]
irb(main):053:0> a1 + a2 - (a1 & a2)
=> ["one", "four"]
```